src/otx/core/data/dataset/object_detection_3d.py

```diff
@@ -68,238 +55,55 @@ def __init__(
-                anno_dict["dimensions"] = [dimension[2], dimension[0], dimension[1]]
-            original_kitti_format = self._reformate_for_kitti_metric(original_kitti_format)
-        # decode labels for training
-        inputs, targets, ori_img_shape = self._decode_item(
-            PILImage.fromarray(image),
-            entity.annotations,
-            calib,
-        )
-        # normilize image
-        inputs = self._apply_transforms(torch.as_tensor(inputs, dtype=torch.float32))
-        return Det3DDataEntity(
-            image=inputs,
+        image, ori_img_shape = self._get_img_data_and_shape(image)
+        calib = self.get_calib_from_file(entity.attributes["calib_path"])
+        annotations_copy = deepcopy(entity.annotations)
+        original_kitti_format = [obj.attributes for obj in annotations_copy]
+
+        # decode original kitti format for metric calculation
+        for i, anno_dict in enumerate(original_kitti_format):
+            anno_dict["name"] = (
+                self.label_info.label_names[annotations_copy[i].label]
+                if self.subset_type != "train"
+                else annotations_copy[i].label
+            )
+            anno_dict["bbox"] = annotations_copy[i].points
+            dimension = anno_dict["dimensions"]
+            anno_dict["dimensions"] = [dimension[2], dimension[0], dimension[1]]
+        original_kitti_format = self._reformate_for_kitti_metric(original_kitti_format)
+
+        entity = Det3DDataEntity(
+            image=image,
             img_info=ImageInfo(
                 img_idx=index,
-                img_shape=inputs.shape[1:],
-                ori_shape=ori_img_shape,  # TODO(Kirill): curently we use WxH here, make it HxW
+                img_shape=ori_img_shape,
+                ori_shape=ori_img_shape,
                 image_color_channel=self.image_color_channel,
                 ignored_labels=[],
             ),
-            boxes=tv_tensors.BoundingBoxes(
-                targets["boxes"],
-                format=tv_tensors.BoundingBoxFormat.XYXY,
-                canvas_size=inputs.shape[1:],
-                dtype=torch.float32,
-            ),
-            labels=torch.as_tensor(targets["labels"], dtype=torch.long),
-            calib_matrix=torch.as_tensor(calib, dtype=torch.float32),
-            boxes_3d=torch.as_tensor(targets["boxes_3d"], dtype=torch.float32),
-            size_2d=torch.as_tensor(targets["size_2d"], dtype=torch.float32),
-            size_3d=torch.as_tensor(targets["size_3d"], dtype=torch.float32),
-            depth=torch.as_tensor(targets["depth"], dtype=torch.float32),
-            heading_angle=torch.as_tensor(
-                np.concatenate([targets["heading_bin"], targets["heading_res"]], axis=1),
-                dtype=torch.float32,
-            ),
+            boxes=np.zeros((self.max_objects, 4), dtype=np.float32),
+            labels=np.zeros((self.max_objects), dtype=np.int8),
```

Comment:
`Det3DDataEntity` get torch but numpy as passed. If numpy needs to be passed, then could you update type hint of `Det3DDataEntity`? And could you revmoe `# mypy: ignore-errors` in this file?

Reply:
Agree, but the problem that the decoding transform and all related functions expect numpy passed. However, the Det3DDataEntity intended to be with torch tensors for training. I will try to change this behavior somehow

Reply:
Done https://github.com/openvinotoolkit/training_extensions/pull/4046/commits/ae60b71090bdc8f7e00776055de3f2c7e9684831